pkg/cluster/internal/create/create.go

```diff
@@ -155,7 +155,17 @@ func Cluster(logger log.Logger, p providers.Provider, opts *ClusterOptions) erro
 		return nil
 	}
 
-	if err := kubeconfig.Export(p, opts.Config.Name, opts.KubeconfigPath); err != nil {
+	// try exporting kubeconfig with backoff for locking failures
+	// TODO: factor out into a public errors API w/ backoff handling?
+	// for now this is easier than coming up with a good API
+	var err error
+	for _, b := range []time.Duration{time.Millisecond, time.Millisecond * 50, time.Millisecond * 100} {
```

Comment:
these times are fairly arbitrary, but a strict improvement over the current behavior.

we don't need to do any significant computation during this time, just lock the files, read them in, write out the patched file with the kubeconfig merged in.

Reply:
it would probably be best to not retry for certain classes of failure, but this is much simpler to get implemented, and the harm is minimal, it will take slightly longer to fail if it's e.g. a permissions issue.